lua/octo/init.lua

```diff
@@ -768,31 +768,42 @@ function M.show_summary()
     return
   end
 
-  local _, repo2 = pcall(api.nvim_buf_get_var, 0, "repo")
-  if not repo2 then
+  local _, current_repo = pcall(api.nvim_buf_get_var, 0, "repo")
+  if not current_repo then
     return
   end
 
   local repo, number
   local cursor = api.nvim_win_get_cursor(0)
   local current_line = vim.fn.getline(".")
-  local different_repo_issue_pattern = "%s([^ /]+/[^ #]+)#(%d+)%s"
+  local different_repo_issue_pattern = "%s([^/]+/[^#]+)#(%d+)%s"
   local same_repo_issue_pattern = "%s#(%d+)%s"
-  local start_col1, end_col1, repo1, number1 = current_line:find(different_repo_issue_pattern)
+  local url_issue_pattern = "[htps]+://.*github.com/([^/]+/[^/]+)/[pulisue]+/(%d+)"
```

Comment:
I think you can pull these variables to the outside of the function. This seems to be a "static" pattern which does not alternate for each run of the function. There is no need to re-allocate memory for this variable each time the function runs. It will help to keep the function more clean by removing what is not part of its behavior or to improve readability.